drivers/timer/ite_it8xxx2_timer.c

```diff
@@ -224,7 +226,7 @@ void sys_clock_set_timeout(int32_t ticks, bool idle)
 	/* Disable event timer */
 	IT8XXX2_EXT_CTRLX(EVENT_TIMER) &= ~IT8XXX2_EXT_ETXEN;
 
-	if (ticks == K_TICKS_FOREVER) {
+	if (ticks == EVENT_TIMER_FOREVER) {
```

Comment:
By the discussion above, this would not happen for the reason it is supposed to catch.  The K_TICKS_FOREVER/K_FOREVER that is passed started as an int64_t and may convert to 0xFFFF FFFF or 0x7FFF FFFF depending on the tool chain when it casts to a uint32_t. The new defined value will be set to 0x7FFF FFFF.  So with some tool chains this may work and others it may not. This seems wrong.

The system already has a k_ticks_t type that it understands and changes the type based on the timer CONFIG_TIMEOUT_64BIT being used for the system. Why not use that as the ticks parameter type. Doing this will cause the value to always be represented correctly as a value or as FOREVER.

Reply:
I correct this PR, I think sys_clock_set_timeout() can handle both forever K_TICKS_FOREVER and INT_MAX case now.